### ISSUE TITLE
completionBlock in addDefaultHandlerForMethod not being fully executed on Xcode 7 beta 6  

### ISSUE THREAD

**Issue by duncandee** [swisspol/GCDWebServer#205]
Hi 

This seems to be a issue in the latest Xcode Beta (works in  Xcode 7 beta 5)

When calling the `completionBlock()` within `addDefaultHandlerForMethod` it seem that data is never written back. Tried using the example below and tried using safari on mac with the URL below. 

**Example:**
```swift
class TestWebServer: NSObject, GCDWebServerDelegate{
  
  private let webServer = GCDWebServer()
  private let urlSession = NSURLSession(configuration: NSURLSessionConfiguration.defaultSessionConfiguration())

  override init() {
    super.init()
    GCDWebServer.setLogLevel(0)
    webServer.delegate = self
    
    webServer.addDefaultHandlerForMethod("GET", requestClass: GCDWebServerDataRequest.self) {
      (request, completionBlock) -> Void in
      completionBlock(GCDWebServerDataResponse(statusCode: 404))
    }
    
    webServer.startWithPort(7777, bonjourName: nil)
  }
  
  func webServerDidStart(server: GCDWebServer!) {
    callLocalHost()
  }
  
  
  func callLocalHost(){
    print("//1 callLocalHost")

    let urlRequest = NSURLRequest(URL: NSURL(string: "http://localhost:7777")!)
    
    let dataTask = self.urlSession.dataTaskWithRequest(urlRequest) {
      data, response, error in
      
      print("//3 callLocalHost: complete \n\n \(error)\n")
    }
    
    print("//2 callLocalHost start")
    dataTask.resume()
  }
}
```
**Log:**
```
[DEBUG] Did open IPv4 listening socket 6
[DEBUG] Did open IPv6 listening socket 7
[INFO] GCDWebServer started on port 7777 and reachable at http://10.1.56.171:7777/
//1 callLocalHost
//2 callLocalHost start
[DEBUG] Did open connection on socket 13
[DEBUG] Did connect
[DEBUG] Did start background task
[DEBUG] Connection received 188 bytes on socket 13
[DEBUG] Connection on socket 13 preflighting request "GET /" with 188 bytes body
[DEBUG] Connection on socket 13 processing request "GET /" with 188 bytes body
[DEBUG] Connection sent 104 bytes on socket 13
//3 callLocalHost: complete 

 Optional(Error Domain=NSURLErrorDomain Code=-1001 "The request timed out." UserInfo={NSUnderlyingError=0x7fd3f0c7d5a0 {Error Domain=kCFErrorDomainCFNetwork Code=-1001 "(null)" UserInfo={_kCFStreamErrorCodeKey=-2102, _kCFStreamErrorDomainKey=4}}, NSErrorFailingURLStringKey=http://localhost:7777/, NSErrorFailingURLKey=http://localhost:7777/, _kCFStreamErrorDomainKey=4, _kCFStreamErrorCodeKey=-2102, NSLocalizedDescription=The request timed out.})
```

**Comment by swisspol** [comment]
The log indicates that data is written back:

    [DEBUG] Connection sent 104 bytes on socket 13

I'm not sure why that would be an issue with GCDWebServer though considering it works with previous versions of Xcode and that code has not changed in a year.

Maybe it's related to https://developer.apple.com/library/prerelease/ios/technotes/App-Transport-Security-Technote/?

Or maybe it's an issue with Swift, which continuously changes, and you should try an Obj-C version. Try Googling to see if other people have similar problems.

**Comment by duncandee** [comment]
Hey

Thanks for the reply. Ill try a sample app in Obj-C. I have logged a bug report with apple so I hope they fix it before the GM drops :( 

Thanks for your time, BTW great product.

**Comment by duncandee** [comment]
I have tried the Obj-C approach and it works as expected. So it seems to be a issue with swift.

**Comment by swisspol** [comment]
Thanks for following up! Looks like nothing can be done on GCDWebServer side here.

**Comment by duncandee** [comment]
Apple has fixed this in Xcode 7.1 Beta 2

**Comment by swisspol** [comment]
Thanks for the update, this is great news :+1:
